src/nnet3/nnet-optimize.cc

```diff
@@ -878,6 +938,12 @@ const NnetComputation* CachingOptimizingCompiler::Compile(
       checker.Check();
     }
     computation_.ComputeCudaIndexes();
+    UpdateCache(true);
+  } else {
+    // if found, get computation and update access queue
+    request_ = *(cit->first);
+    computation_ = *(cit->second.first);
```

Comment:
There is no reason to have the variables 'computation_' and 'request_', they are just redundant copies.  Just return the pointer to the value in the map.